src/helpers/pull-helpers/has-collaborator-converted.ts

```diff
@@ -0,0 +1,31 @@
+import { Context } from "../../types";
+
+export async function hasCollaboratorConvertedPr(context: Context<"pull_request.opened" | "pull_request.ready_for_review">) {
```

Comment:
Maybe I'm missing something but instead of checking the timeline, you could just compare `payload.sender.login` with `payload.pull_request.user.login`, so this will run when the PR is opened or PR author converts it to `ready for review`